### PR TITLE
Add e2e to for Lighthouse HA Scenario

### PR DESCRIPTION
**Pull request by aswinsuryan**
* Tests when active pods are  present for service only in one
cluster while services are available in both the clusters.

Signed-off-by: Aswin Surayanarayanan <asuryana@redhat.com>

### CODE REVIEW

File: test/e2e/discovery/service_discovery.go [hunk 1]
```diff
@@ -193,6 +199,56 @@ func RunServicesPodAvailabilityTest(f *lhframework.Framework) {
...
+
+	f.AwaitServiceExportedStatusCondition(framework.ClusterA, nginxServiceClusterA.Name, nginxServiceClusterA.Namespace)
+
+	if svc, err := f.GetService(framework.ClusterA, nginxServiceClusterA.Name, nginxServiceClusterA.Namespace); err == nil {
+		nginxServiceClusterA = svc
+		f.AwaitServiceImportIP(framework.ClusterA, nginxServiceClusterA)
+	}
+
+	f.SetNginxReplicaSet(framework.ClusterB, 0)
+	verifyServiceIpWithDig(f.Framework, framework.ClusterA, nginxServiceClusterA, netshootPodList, checkedDomains, true)
```
Comment (tpantelis) [inline test/e2e/discovery/service_discovery.go:245]: Perhaps we should do `verifyServiceIpWithDig` a second time to ensure the first query didn't return clusterA's service IP due to round-robin?

File: test/e2e/discovery/service_discovery.go [hunk 2]
```diff
@@ -46,6 +46,12 @@ var _ = Describe("[discovery] Test Service Discovery Across Clusters", func() {
 			RunServicesPodAvailabilityTest(f)
 		})
 	})
+
+	When("there is active pods for a service in only one cluster", func() {
```
Comment (tpantelis) [inline test/e2e/discovery/service_discovery.go:50]:
```suggestion
	When("there are active pods for a service in only one cluster", func() {
```

File: test/e2e/discovery/service_discovery.go [hunk 3]
```diff
@@ -46,6 +46,12 @@ var _ = Describe("[discovery] Test Service Discovery Across Clusters", func() {
...
+		It("should not resolve the service service without active pods", func() {
```
Comment (tpantelis) [inline test/e2e/discovery/service_discovery.go:51]:
```suggestion
		It("should not resolve the service on the cluster without active pods", func() {
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: test/e2e/discovery/service_discovery.go (hunk 1 above)
Comment (vthapar) [reply test/e2e/discovery/service_discovery.go:245]: That is not needed. `verifyServiceIpWithDig` will keep retrying till it returns the IP we asked for.

File: test/e2e/discovery/service_discovery.go [hunk 4]
```diff
@@ -193,6 +199,56 @@ func RunServicesPodAvailabilityTest(f *lhframework.Framework) {
...
+	}
+
+	f.SetNginxReplicaSet(framework.ClusterB, 0)
+	verifyServiceIpWithDig(f.Framework, framework.ClusterA, nginxServiceClusterA, netshootPodList, checkedDomains, true)
+
+	f.SetNginxReplicaSet(framework.ClusterA, 0)
+	f.SetNginxReplicaSet(framework.ClusterB, 2)
+	verifyServiceIpWithDig(f.Framework, framework.ClusterA, nginxServiceClusterB, netshootPodList, checkedDomains, true)
+}
+
```
Comment (vthapar) [inline test/e2e/discovery/service_discovery.go:251]: Here add a `verifyServiceIpWithDig` for `nginxServiceClusterA` with `shouldContain=false` to confirm it is not returning IP of service with replica size 0.

**Review by vthapar — CHANGES_REQUESTED** [review]

File: test/e2e/discovery/service_discovery.go [hunk 5]
```diff
@@ -193,6 +199,56 @@ func RunServicesPodAvailabilityTest(f *lhframework.Framework) {
...
+	if svc, err := f.GetService(framework.ClusterA, nginxServiceClusterA.Name, nginxServiceClusterA.Namespace); err == nil {
+		nginxServiceClusterA = svc
+		f.AwaitServiceImportIP(framework.ClusterA, nginxServiceClusterA)
+	}
+
+	f.SetNginxReplicaSet(framework.ClusterB, 0)
+	verifyServiceIpWithDig(f.Framework, framework.ClusterA, nginxServiceClusterA, netshootPodList, checkedDomains, true)
+
+	f.SetNginxReplicaSet(framework.ClusterA, 0)
+	f.SetNginxReplicaSet(framework.ClusterB, 2)
```
Comment (vthapar) [inline test/e2e/discovery/service_discovery.go:248]:
```suggestion
```
This is not needed, doesn't add any value.

**Review by vthapar — CHANGES_REQUESTED** [review]

File: test/e2e/discovery/service_discovery.go (hunk 1 above)
Comment (aswinsuryan) [reply test/e2e/discovery/service_discovery.go:246]: Modfied the test to check the false(to check if the custerIP is removed) condition too

File: test/e2e/discovery/service_discovery.go (hunk 2 above)
Comment (aswinsuryan) [reply test/e2e/discovery/service_discovery.go]: Done.

File: test/e2e/discovery/service_discovery.go (hunk 3 above)
Comment (aswinsuryan) [reply test/e2e/discovery/service_discovery.go]: Done.

File: test/e2e/discovery/service_discovery.go (hunk 5 above)
Comment (aswinsuryan) [reply test/e2e/discovery/service_discovery.go]: Done.

File: test/e2e/discovery/service_discovery.go (hunk 4 above)
Comment (aswinsuryan) [reply test/e2e/discovery/service_discovery.go:253]: Done.

**Review by tpantelis — APPROVED** [review]

**Review by vthapar — APPROVED** [review]

**Pull request merged**